File: api/schemas/recipe.py
from datetime import datetime

from sqlmodel import SQLModel

from api.models import User
from api.schemas.ingredient import IngredientDetail


class RecipeSlim(SQLModel):
    id: int | None
    name: str
    description: str
    instructions: str
    notes: str | None
    created_on: datetime
    public: bool


class RecipeDetail(SQLModel):
    id: int | None
    name: str
    description: str
    instructions: str
    notes: str | None
    created_on: datetime
    public: bool

    created_by: User
    ingredients: list["IngredientDetail"]


class RecipeCreate(SQLModel):
    name: str
    description: str
    instructions: str
    notes: str | None
    created_on: datetime
    public: bool

    created_by: User


class RecipeUpdate(SQLModel):
    name: str | None
    description: str | None
    instructions: str | None
    notes: str | None
    created_on: datetime | None
    public: bool | None

    created_by: User | None

File: src/types/Recipe.ts
import type { IngredientSlim } from "./Ingredient";
import type { UploadSlim } from "./Upload";
import type { UserResponse } from "./User";

export interface RecipeSlim {
  id: number;
  name: string;
  description: string;
  instructions: string;
  notes?: string;
  created_on: string;
  created_by_id: number;
  public: boolean;
  prep_time?: number;
  cover_image_id?: number;
}

export interface RecipeDetail extends RecipeSlim {
  created_by: UserResponse;
  ingredients: IngredientSlim[];
  cover_image?: UploadSlim;
}

export interface RecipeDashboard extends RecipeSlim {
  cover_image?: UploadSlim;
}

export interface RecipeCreate {
  name: string;
  description: string;
  instructions: string;
  notes?: string;
  created_on: string;
  public: boolean;
  prep_time?: number;
  cover_image_id?: number;
}
